fe/fe-core/src/main/java/com/starrocks/persist/gson/RuntimeTypeAdapterFactory.java

```diff
@@ -361,11 +361,17 @@ public R read(JsonReader in) throws IOException {
                     labelJsonElement = jsonElement.getAsJsonObject().remove(typeFieldName);
                 }
 
+                //1. if the typeFieldName is specified in the json string, then use this class.
+                //2. if default class if specified, then use the default class.
+                //3. if the type specified (the second param of GSON.fromJson) is in the register list,
+                //   then use this class.
                 String label;
                 if (labelJsonElement != null) {
                     label = labelJsonElement.getAsString();
                 } else if (defaultLabel != null) {
                     label = defaultLabel;
+                } else if (subtypeToLabel.containsKey(type.getRawType())) {
```

Comment:
you should also remove `.registerSubtype(InsertTxnCommitAttachment.class, InsertTxnCommitAttachment.class.getSimpleName(), true)` in this pr https://github.com/StarRocks/starrocks/pull/23833/files ?

Reply:
done